src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/EncryptedXml.cs

```diff
@@ -204,46 +204,46 @@ private byte[] GetCipherValue(CipherData cipherData)
                 if (cipherData.CipherReference.Uri.Length == 0)
                 {
                     // self referenced Uri
-                    string baseUri = (_document == null ? null : _document.BaseURI);
+                    string? baseUri = (_document == null ? null : _document.BaseURI);
                     TransformChain tc = cipherData.CipherReference.TransformChain;
                     if (tc == null)
                     {
                         throw new CryptographicException(SR.Cryptography_Xml_UriNotSupported);
                     }
-                    decInputStream = tc.TransformToOctetStream(_document, _xmlResolver, baseUri);
+                    decInputStream = tc.TransformToOctetStream(_document!, _xmlResolver!, baseUri!);
```

Comment:
TransformToOctetStream is internal.  If every caller is having to `!` away all their arguments, it seems easier to just declare it as null-accepting.

Reply:
This is a tricky one. I'm concentrating on `baseUri`, e.g. when it's set on line 207. It's marked as nullable because there's a check for `_document == null ? null : _document.BaseURI`. In reality, `_document` is never null. So if that expression to check for null is removed, then `baseUri` could be declared as non-null. So that'd be one `!` we could get rid of to make the code easier to follow.
But, I was unsure if I should remove the `_document == null` part of the expression. What do you think?

Reply:
But as far as the `_xmlResolver` field is concerned: it's explicitly set to null on construction, and can be mutated via the public `Resolver` property, but I cannot see anywhere in the type where it physically checked for null.
So, if I make the `resolver` parameter null-accepting, I then get more ambiguities further on, e.g on line 85 where we call `Utils.PreProcessStreamInput`, which accepts a non-null `resolver`. 

What I've done in light of this is make `xmlResolver`  null-accepting, including in the constructors of `CanonicalXml` (which seems to be the right thing to do, as there is an explicit null check for `inputStream` or `document`, but not for `resolver`).

The changes related to this are in their own commit for an easy revert if necessary.